docker/linux/scripts/configure-tentacle.sh

```diff
@@ -103,7 +107,7 @@ function configureTentacle() {
 	if [[ ! -z "$ServerPort" ]]; then
 		tentacle configure --instance "$instanceName" --noListen "True"
 	else
-		tentacle configure --instance "$instanceName" --port $ListeningPort --noListen "False"
+		tentacle configure --instance "$instanceName" --port $defaultListeningPort --noListen "False"
```

Comment:
If a `ListeningPort` wasn't provided before for a Linux container it would fail here with an unhelpful error message. It should default to `10933`.

Reply:
It looks like previously the port could be specified by setting `$ListeningPort`, is there a way to customise the port now?

Reply:
This line sets the port in the Tentacle's configuration, but when it does the `register-with` it [overrides it](https://github.com/OctopusDeploy/OctopusTentacle/pull/350/files/569e20ed37693d933ec5c5a61d2ee626e63e934e#diff-3d9f587a7deebe13ba8499e4940aaa4d4b4cb9a1c7181e73ec0eaf86a9f2bf4bR181) with `$ListeningPort` if it exists and is different:
```
		if [[ ! -z "$ListeningPort" && "$ListeningPort" != $defaultListeningPort ]]; then
			ARGS+=('--tentacle-comms-port' $ListeningPort)
		fi
```
It does this a similar thing in the Windows version already:
https://github.com/OctopusDeploy/OctopusTentacle/blob/main/docker/windows/Scripts/configure-tentacle.ps1#L47
https://github.com/OctopusDeploy/OctopusTentacle/blob/main/docker/windows/Scripts/configure-tentacle.ps1#L208

Reply:
Will setting `--tentacle-comms-port` alter the Tentacle configuration?

Reply:
Is it a container thing? The Tentacle should always be installed on port 10933, but may be mapped to another external port that the Octopus Server should use for comms?

Reply:
> The Tentacle should always be installed on port 10933, but may be mapped to another external port that the Octopus Server should use for comms?

I'm _pretty sure_ this is the case. Eg. in this image:
https://user-images.githubusercontent.com/11970672/149034313-3fd39afe-3616-4897-b65c-4e91a6632319.png

I set up the container with `-p 10934:10933` - meaning that port 10934 on the host maps to port 10933 on the Docker instance. I added `--env ListeningPort="10934"` which tells Server to use 10934 for comms.

Reply:
> Will setting --tentacle-comms-port alter the Tentacle configuration?

I don't think so. It is a bit confusing that it gets configured like this, maybe a better way would be to set `$ListeningPort` to `$defaultListeningPort` at the start of the script if it doesn't have a value and then use `$ListeningPort` for everything.

Reply:
Using just `$ListeningPort` as the linux image was, we suspect the docker ports must line up for registration to work:

```
-p 10933:10933
$ListeningPort = 10933
configure --port $ListeningPort
register-with --tentacle-comms-port $ListeningPort
```

If we start using the default port, the user will be able to remap the docker ports (seems like a good idea):
``
-p 10934:10933
$ListeningPort = 10934
configure --port $DefaultPort
register-with --tentacle-comms-port $ListeningPort
```

Suspected problem with changing it is that it will break anyone currently using a `$ListeningPort` that is not 10933. They would need to have their docker ports matching to make the registration work. eg: `-p 10934:10934`

Reply:
We're going to revert this change for now. Slack thread for discussion: https://octopusdeploy.slack.com/archives/C01J6U3MHJ4/p1641958569075100